potassco/reify_utils.h

```diff
@@ -0,0 +1,107 @@
+// {{{ MIT License
+
+// Copyright 2017 Roland Kaminski
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+
+// }}}
+
+
+#pragma once
+
+#include <potassco/basic_types.h>
+
+#include <iostream>
+#include <vector>
+
+namespace Potassco {
+
+template <typename T>
+void printValue(std::ostream& out, const T& value) { out << value; }
+
+template <typename T, typename U>
```

Comment:
General remark: It seems that the only relevant std::pair that is used could be replaced by the existing `WeightLit` type. Please consider switching to this type here, in Hash, and in reify.h